src/utils/fillwidth.ts

```diff
@@ -47,19 +49,24 @@ const fillwidthDimensions = (items, containerWidth, gutter = 10, targetHeight) =
     }
   })
 
-  // If the total width difference is less than 0, it is larger than the container
-  // so we need to scale down. If not, scale up.
-  let dir = widthDiff(dimensions) < 0 ? -1 : 1
+  // If the total width difference is too small or negative we need to scale down. If not, scale up.
+  let dir = widthDiff(dimensions) < 1 ? -1 : 1
 
   // Keep looping until we get an acceptable width difference
-  while (true) {
+  let count = 0
+  while (widthDiff(dimensions) <= 1) {
     for (let img of dimensions) {
       resizeHeight(img, dir)
       if (widthDiff(dimensions) > 1) {
         break
       }
     }
-    if (widthDiff(dimensions) > 1) {
```

Comment:
Putting the first check _after_ resizing at least once, means that the containers could get resized too much. Instead we now check it at the start of the loop, possibly never resizing if the difference was already ok to begin with.